pkg/cluster/sync.go

```diff
@@ -950,13 +948,21 @@ func (c *Cluster) syncRoles() (err error) {
 	// update pgUsers where a deleted role was found
 	// so that they are skipped in ProduceSyncRequests
 	for _, dbUser := range dbUsers {
-		if originalUser, exists := deletedUsers[dbUser.Name]; exists {
-			recreatedUser := c.pgUsers[originalUser]
+		originalUsername, foundDeletedUser := deletedUsers[dbUser.Name]
+		// check if original user does not exist in dbUsers
+		_, originalUserAlreadyExists := dbUsers[originalUsername]
+		if foundDeletedUser && !originalUserAlreadyExists {
```

Comment:
With this PR the operator can revoke LOGIN from users but is [allowed to fail the renaming](https://github.com/zalando/postgres-operator/pull/2072/files#r991361190). This means, both users with and without suffix cannot login anymore. When the member joins the team again [altering the deprecated user is skipped](https://github.com/zalando/postgres-operator/pull/2072/files#r991363521), but LOGIN should be granted back to the role without the suffix. [ProduceSyncRequests](https://github.com/zalando/postgres-operator/blob/84fe38a06908aa8adbe580041c133db84faf7121/pkg/util/users/users.go#L70) can do this, but only if we do not mark the user as deleted here.